### PR TITLE
Allow the unicycle to move sideways

### PR DESCRIPTION
**Pull request by S-Dafarra**
This PR adds a new controller for the Unicycle, permitting side steps.

### CODE REVIEW

File: include/UnicycleDirectController.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ *
+ */
+
+#ifndef UNICYCLEDIRECTCONTROLLER_H
+#define UNICYCLEDIRECTCONTROLLER_H
+
+#include "UnicycleBaseController.h"
+
+class UnicycleDirectController : public UnicycleBaseController
```
Comment (GiulioRomualdi) [inline include/UnicycleDirectController.h:13]: If I correctly understand this allows the robot to move in the y direction (i.e. it brakes the holonomic constraint) if this is the case I would avoid calling it `UnicycleDirectController`

**Review by GiulioRomualdi — COMMENTED** [review]
One minor comment

File: include/UnicycleDirectController.h (hunk 1 above)
Comment (S-Dafarra) [reply include/UnicycleDirectController.h:13]: Nope, this one controller of the unicycle that allows the user to set the unicycle velocities directly. The other controller is the ``PersonFollowing`` controller, where the user sets a desired point to follow.

in order to move sideways I have extended the dynamics of the unicycle in https://github.com/robotology/unicycle-footstep-planner/blob/357445516d1504fe809fdf2ef5696b39beee9c9f/src/ControlledUnicycle.cpp#L55-L56, assuming to have an additional control input. The ``PersonFollowing`` controller simply sets this control input always to zero.

**Comment by GiulioRomualdi** [comment]
Ok! Perfect feel free to merge it. Could you please release the software? It will allow us to release the walking as well.

**Comment by S-Dafarra** [comment]
> Ok! Perfect feel free to merge it. Could you please release the software? It will allow us to release the walking as well.

Sure!

**Comment by S-Dafarra** [comment]
> Ok! Perfect feel free to merge it. Could you please release the software? It will allow us to release the walking as well.

See https://github.com/robotology/unicycle-footstep-planner/releases/tag/v0.5.0

**Pull request merged**